### PR TITLE
#161254622 Rate articles on Authors Heaven

### PR DESCRIPTION
**Pull request by wrotich**
#### What does this PR do?
```This PR merges the rate article branch with develop ```
#### Description of Task to be completed?
```allow logged in users to rate articles on Authors Haven```
```allow users to view average rating of an article ```
```Redesign the view article UI```
#### How should this be manually tested?
```Clone this repository```
- git clone https://github.com/andela/ah-orcas-frontend.git
```Change the working directory to the project folder```
- cd ah-orcas-frontend
```Checkout to the appropriate branch```
- git checkout ft-rate-an-article-161254622
```Install the required dependencies and start the server```
- npm install
- npm start
```You should be able to see the star ratings when you view a single article, i.e. average rating. Also, you will be able to rate a given article provided you are logged in```
#### Any background context you want to provide?
```Having ratings on a given article helps in getting the statistics that can be useful when collecting data on which topic is loved by the users ```
#### What are the relevant pivotal tracker stories?
[#161254622](https://www.pivotaltracker.com/story/show/161254622)
#### Screenshots (if appropriate)
screenshot 2018-12-20 at 13 31 27

screenshot 2018-12-20 at 14 49 40

#### Checklist:

- [X] My code follows the style guidelines of this project
- [ ] At least 2 people have reviewed my PR
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] My PR has one commit.

### CODE REVIEW

File: src/components/ArticleView/App.js [hunk 1]
```diff
@@ -6,29 +6,36 @@ import './App.scss';
...
+  async componentDidMount() {
     const slug = localStorage.getItem('slug');
-    const { singleArticle } = this.props;
+    const { singleArticle } = await this.props;
     singleArticle(slug);
   }
 
   render() {
-    const { results } = this.props;
+    const { results, rates } = this.props;
```
[the remark this answers is not in the document]
Comment (wrotich) [reply src/components/ArticleView/App.js:21]: Resolved

**Review by evamaina — APPROVED** [review]
LGTM

**Review by Gidraf — APPROVED** [review]
LGTM

File: src/actions/articles.action.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import axios from 'axios';
 import { GET_ARTICLES, ADD_ARTICLE, VIEW_ARTICLE, DELETE_ARTICLE, DELETE_ARTICLE_FAILURE, DELETE_ARTICLE_SUCCESS, UPDATE_ARTICLE } from '../constants/articles';
 import baseUrl from '../constants/url';
+import { avgRate } from './ratings.actions';
 
 const token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1NDUzMTQ4MTMsImlhdCI6MTU0NDcxMDAxMywiaWQiOjUxfQ.qB3oqQNnoeoYnaTRIp9h4TXAmARQLfUjVVZzUpjAfCw';
```
Comment (TheSteelGuy) [inline src/actions/articles.action.js]: pass this token from the actual login

**Review by TheSteelGuy — CHANGES_REQUESTED** [review]

**Review by TheSteelGuy — APPROVED** [review]

**Pull request merged**